File: 백준/Bronze/2501. 약수 구하기/약수 구하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        List<Integer> factorList = new ArrayList<>();

        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        for (int i = 1; i <= n; i++) {
            if (n % i == 0) {
               factorList.add(i);
            }
        }
        Collections.sort(factorList);

        if (k > factorList.size()) {
            System.out.println(0);
        }else {
            System.out.println(factorList.get(k-1));
        }

    }
}

File: 백준/Bronze/2501. 약수 구하기/약수 구하기.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val input = br.readLine().split(" ").map { it.toInt() }
    
    val N = input[0]
    val K = input[1]
    
    val list = mutableListOf<Int>()
    
    for(i in 1..N){
        if(N % i == 0) {
            list.add(i)
        }
    }
    
    if(list.size < K) {
        println(0)
    }else {
        println(list[K-1])
    }
}